src/ncp/ncp_host.hpp

```diff
@@ -0,0 +1,82 @@
+/*
+ *  Copyright (c) 2024, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file includes definitions of OpenThead Controller for NCP.
+ */
+
+#ifndef OTBR_AGENT_NCP_HOST_HPP_
+#define OTBR_AGENT_NCP_HOST_HPP_
+
+#include "lib/spinel/coprocessor_type.h"
+
+#include "thread_controller.hpp"
+#include "common/mainloop.hpp"
+
+namespace otbr {
+namespace Ncp {
+
+class NcpHost : public MainloopProcessor, public ThreadController
```

Comment:
Why this is not named "NcpController" or "NcpThreadControler" which is more clearer that this is a "ThreadController" implementation.

Reply:
As discussed offline, since currently there is already `RcpHost`, we will still use the name `NcpHost`. While it makes sense that we also change the name of interface `ThreadController` to `ThreadHost`. Will use a seperate PR to update the name of the interface.

Reply:
A PR has been submitted: https://github.com/openthread/ot-br-posix/pull/2332/